daffodil-core/src/main/scala/org/apache/daffodil/dsom/SchemaSet.scala

```diff
@@ -472,7 +472,12 @@ final class SchemaSet(
       }
 
       val newNS = matchingDVs.head.namespace
-      val newBinding = Binding(v.varQName.local, Some(newNS), v.varValue)
+      val newBinding = try {
+        Binding(v.varQName.local, Some(newNS), v.varValue)
+      } catch {
+        case qupe: QNameUndefinedPrefixException => this.SDE("Undefined QName prefix in externally defined variable %s: %s", v.varQName, qupe.getMessage)
```

Comment:
Same here. You shouolnd't have to have two cases here. Just one for a base-class/base trait.